src/gtk/fm-dnd-dest.c

```diff
@@ -984,6 +984,7 @@ GdkDragAction fm_dnd_dest_get_default_action(FmDndDest* dd,
             /* retrieve the source files */
             gtk_drag_get_data(dd->widget, drag_context, target, time(NULL));
             dd->waiting_data = TRUE;
+            action = drag_context->suggested_action;
```

Comment:
Could you explain, please, what kind of problem this change solves? This is intermediate case. And also your change is incompatible with GTK+ 3.0, BTW.